packages/stream-caip10-link-handler/src/caip10-link-handler.ts

```diff
@@ -153,7 +157,9 @@ export class Caip10LinkHandler implements StreamHandler<Caip10Link> {
     state: StreamState
   ): Promise<StreamState> {
     // TODO: Assert that the 'prev' of the commit being applied is the end of the log in 'state'
-    const proof = (await context.ipfs.dag.get(commit.proof, { timeout: IPFS_GET_TIMEOUT })).value
+    const proof = StreamUtils.isExpandedCommit(commit)
```

Comment:
Can this ever be false?  Would be nice not to have to have the fallback logic to loading the proof here.  It makes it hard to know what the actual control flow is going to be

Reply:
same question for the above checks for signed commits.  Can we just have a single type that is used throughout the StreamtypeHandlers?